.github/workflows/integrationtests.yaml

```diff
@@ -100,3 +106,49 @@ jobs:
 
       - name: Daps Integration Tests
         run: ./gradlew -p extensions/iam/daps test -DincludeTags="DapsIntegrationTest"
+
+  Check-Cloud-Environments:
+    environment: Azure-dev
+    runs-on: ubuntu-latest
+    steps:
+      - id: has-azure
+        env:
+          HAS_AZURE: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        if: "${{ env.HAS_AZURE != '' }}"
+        run: echo "::set-output name=defined::true"
+    outputs:
+      has-azure: ${{ steps.has-azure.outputs.defined }}
+
+  Azure-Cloud-Integration-Test:
+    needs: [ Check-Cloud-Environments ]
+    if: needs.Check-Cloud-Environments.outputs.has-azure == 'true'
+    environment: Azure-dev
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: 'Az CLI login'
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: "Fetch Terraform outputs"
+        run: printf "$TERRAFORM_OUTPUTS" > resources/azure/testing/terraform_outputs.json
```

Comment:
won't that leak potentially sensitive information, for example `tenant_id` or `client_id`?

Reply:
These are not sensitive values. In any case, they are not more exposed because they are on the filesystem than because they are passed to test code.

Reply:
if they are stored as repo secrects, they would be obscured in log output, right? Since the `client_id` is half of the credentials package, i'd very much regard it as sensitive.

Reply:
The only reason client_id is stored as a secret is that GitHub Actions does not propose a way to store repo action settings as "non-secrets". It is no more sensitive than your GitHub ID is.

In any case, it is not being exposed. It is written to the local file system that is wiped after execution. Just like the actual secrets are written to the local Terraform state file.